### PR TITLE
Factor out subcommands into their own functions.

### PR DESCRIPTION
**Pull request by jsha**
This doesn't change any of the bodies of the subcommands, so no need to
re-review those; just took the bodies wholesale and moved them into
their own functions. One exception: The formatter unwrapped some lines
because by outdenting them, more can fit on a line.

Add `&w=1` to the review URL to ignore whitespace diffs.

### CODE REVIEW

File: facilitator/src/bin/facilitator.rs [hunk 1]
```diff
@@ -547,260 +547,256 @@ fn main() -> Result<(), anyhow::Error> {
...
+}
+
+fn intake_batch(sub_matches: &ArgMatches) -> Result<(), anyhow::Error> {
+    let mut intake_transport = intake_transport_from_args(sub_matches)?;
+
+    // We need the bucket to which we will write validations for the
+    // peer data share processor, which can either be fetched from the
+    // peer manifest or provided directly via command line argument. We
+    // check the manifest argument first because peer-output will always
+    // have at least the default value ".".
```
Comment (tgeoghegan) [inline facilitator/src/bin/facilitator.rs:622]: This comment is out of date now that we've removed all default values, though that's not what this PR is about.

File: facilitator/src/bin/facilitator.rs [hunk 2]
```diff
@@ -547,260 +547,256 @@ fn main() -> Result<(), anyhow::Error> {
...
+        &Uuid::parse_str(sub_matches.value_of("batch-id").unwrap()).unwrap(),
+        &NaiveDateTime::parse_from_str(&sub_matches.value_of("date").unwrap(), DATE_FORMAT)
+            .unwrap(),
+        &mut intake_transport,
+        &mut peer_validation_transport,
+        &mut own_validation_transport,
+        is_first_from_arg(sub_matches),
+    )?;
+    batch_intaker.generate_validation_share()?;
+    Ok(())
```
Comment (tgeoghegan) [inline facilitator/src/bin/facilitator.rs:659]: nit: just do `batch_intaker.generate_validation_share()` and drop the `Ok(())`

File: facilitator/src/bin/facilitator.rs [hunk 3]
```diff
@@ -547,260 +547,256 @@ fn main() -> Result<(), anyhow::Error> {
...
+        StoragePath::from_str(sub_matches.value_of("peer-input").unwrap())?;
+    let peer_identity = sub_matches.value_of("peer-identity");
+
+    let peer_validation_transport = transport_for_path(peer_validation_bucket, peer_identity)?;
+
+    // We need the public keys the peer data share processor used to
+    // sign messages, which we can obtain by argument or by discovering
+    // their specific manifest. peer-public-key and
+    // peer-public-key-identifier have default values, so we must check
+    // peer-manifest-base-url first.
```
Comment (tgeoghegan) [inline facilitator/src/bin/facilitator.rs:709]: This comment is also out of date

**Review by tgeoghegan — APPROVED** [review]
LGTM, though I noticed a few things that were already wrong before this PR. Feel free to ignore those if you want to merge and move on.

**Comment by jsha** [comment]
I'm going to merge this as-is so we have a nice clean entry in the log, then do a separate followup with those fixes.

**Pull request merged**